test/Approximators/RangeApproximator/randsvd.jl

```diff
@@ -0,0 +1,334 @@
+            approx = RandSVD(compressor, 2, false)
+            @test_logs (:warn,
+                "Compressor with cardinality `Left` being applied from `Right`."
+            ) complete_approximator(approx, A)
+            approx_rec = complete_approximator(approx, A)
+            approx_rec.compressor.cardinality == Left()
+            
+            @test approx_rec.power_its == 2
+            @test approx_rec.rand_subspace == false 
+            @test approx_rec.n_rows == 10
+            @test approx_rec.n_cols == 5
+        end
+        
+    end
+
+    @testset "RandSVD Recipe: rapproximate" begin
+        # By testing the rapproximate function we also test rapproximate!
+        # with power iterations
+        let n_rows = 10,
+            n_cols = 10,
+            compression_dim = 5,
+            cardinality = Right(),
+            A = rand(n_rows, n_cols)
+
+            compressor = TestCompressor(cardinality, compression_dim)
+            approx = RandSVD(compressor, 2, false)
+            approx_rec = rapproximate(approx, A)
+            
+            @test typeof(approx_rec.compressor) == TestCompressorRecipe  
+            # Check that the matrix is orthogonal
+            gram_matrix = approx_rec.U' * approx_rec.U
+            # check that the norm is 1, the diagonal is all 1
+            @test opnorm(gram_matrix) ≈ 1
+            # test that the diagonal is all ones
+            diag_gram_matrix = diag(gram_matrix)
+            for i = 1:compression_dim
+                @test diag_gram_matrix[i] ≈ 1
+            end
+
+            # bound from theorem 9.1 of Halko Martinsson and Tropp is satisfied
+            _,S,_ = svd(A)
+            norm(S[1:5] - approx_rec.S) < norm(S[6:end])^2 + 
+                norm(Diagonal(S[6:end]) * 
+                approx_rec.compressor.op[6:end, :] * 
+                pinv(approx_rec.compressor.op[1:5, :]))^2
+        end
+
+        # Test that compression_dim == n_rows gives a matrix that spans the range of A
+        # with power iterations, this way the approximation is exact
+        let n_rows = 10,
+            n_cols = 10,
+            compression_dim = 10,
+            cardinality = Right(),
+            A = rand(n_rows, n_cols)
+
+            compressor = TestCompressor(cardinality, compression_dim)
+            approx = RandSVD(compressor, 2, false)
+            approx_rec = rapproximate(approx, A)
+            
+            @test typeof(approx_rec.compressor) == TestCompressorRecipe  
+            approx_rec.compressor.cardinality == Right()
+            @test approx_rec.power_its == 2
+            @test approx_rec.rand_subspace == false
+            @test approx_rec.n_rows == 10
+            @test approx_rec.n_cols == 10 
+            # Check that the matrix is orthogonal
+            gram_matrix = approx_rec.U' * approx_rec.U
+            # check that the norm is 1, the diagonal is all 1
+            @test opnorm(gram_matrix) ≈ 1
```

Comment:
This should be moved after you check the diagonals are one, and add a comment that this checks if the off-diagonals are nearly zero.